src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
> Maybe signed_integer_overflow should just be generalized

Except that we probably want to handle float cases as well.

If we want to generalize this, how's this for a proposal:
-- add `is_finite_expression()`, which returns false for expressions that are provably NaN or Inf 
-- disallow simplifications that consider non-finite values to be numerically equal